src/components/Modal/Modal.tsx

```diff
@@ -0,0 +1,97 @@
+import { Dialog, Transition } from '@headlessui/react';
+import { XIcon } from '@heroicons/react/solid';
+import useEscapeKeyDetector from 'hooks/useEscapeKeyDetector';
+import useOutsideClickDetector from 'hooks/useOutsideClickDetector';
+import React, { Fragment } from 'react';
+
+interface ModalProps {
+  content: React.ReactNode;
```

Comment:
@madhav-devslane I believe taking content as a props can cumbersome when this component will be used. Instead this component can take the content in its intrinsic prop `children` so the parent of this component will have to pass the content between the tags.

Reply:
@kush-prof sir Updated : c715bcca2354cc329d010dc37e893d68cbf2a84e